packages/grid/x-data-grid/src/components/panel/GridPanel.tsx

```diff
@@ -25,6 +25,7 @@ export interface GridPanelProps
    */
   classes?: Partial<GridPanelClasses>;
   open: boolean;
+  onPanelClose?: () => void;
```

Comment:
"panel" is not necessary because all props refer to `GridPanel`.

```suggestion
  onClose?: () => void;
```